### PR TITLE
Privacy Activities: Pass Request

### PR DESCRIPTION
**Pull request by SyntaxNode**
Extends the Activities Control infrastructure to pass request information. Implemented Phase 3.5 GPP SID signal as an initial proof of concept for the new request object.

### CODE REVIEW

File: privacy/rules_test.go [hunk 1]
```diff
@@ -14,7 +16,7 @@ func TestComponentEnforcementRuleEvaluate(t *testing.T) {
 		activityResult ActivityResult
 	}{
 		{
-			name: "activity_is_allowed",
+			name: "activity-allowed",
```
Comment (SyntaxNode) [inline privacy/rules_test.go:19]: The `is` is redudant.

File: privacy/scrubber.go [hunk 2]
```diff
@@ -87,11 +88,8 @@ func (scrubber) ScrubRequest(bidRequest *openrtb2.BidRequest, enforcement Enforc
 	var userExtParsed map[string]json.RawMessage
 	userExtModified := false
 
-	var userCopy *openrtb2.User
-	userCopy = ptrutil.Clone(bidRequest.User)
-
-	var deviceCopy *openrtb2.Device
-	deviceCopy = ptrutil.Clone(bidRequest.Device)
+	userCopy := ptrutil.Clone(bidRequest.User)
+	deviceCopy := ptrutil.Clone(bidRequest.Device)
```
Comment (SyntaxNode) [inline privacy/scrubber.go:92]: VS Code "yelled" at me to make this nitpick optimization.

Comment (VeronikaSolovei9) [reply privacy/scrubber.go:97]: Oh so does it actually work? I also had an error that looked like it's not compiling, so I added those types explicitly.

Comment (VeronikaSolovei9) [reply privacy/scrubber.go:97]: Screenshot 2023-09-05 at 5 04 15 PM

File: privacy/activity.go [hunk 3]
```diff
@@ -10,8 +11,8 @@ const (
 	ActivityReportAnalytics
 	ActivityTransmitUserFPD
 	ActivityTransmitPreciseGeo
-	ActivityTransmitUniqueRequestIds
-	ActivityTransmitTids
+	ActivityTransmitUniqueRequestIDs
+	ActivityTransmitTIDs
```
Comment (VeronikaSolovei9) [inline privacy/activity.go:15]: Oh sorry about this...

File: privacy/rules.go [hunk 4]
```diff
@@ -57,3 +61,30 @@ func evaluateComponentType(target Component, componentTypes []string) bool {
...
+	}
+
+	for _, x := range getGPPSID(request) {
+		for _, y := range sid {
+			if x == y {
+				return true
+			}
+		}
+	}
+	return false
```
[the remark this answers is not in the document]
Comment (VeronikaSolovei9) [reply privacy/rules.go:77]: Just a suggestion: we can turn `sid` to map in `activityRulesToEnforcementRules` and then access it like a map?

File: privacy/rules.go [hunk 5]
```diff
@@ -57,3 +61,30 @@ func evaluateComponentType(target Component, componentTypes []string) bool {
...
+	if request.IsPolicies() {
+		return request.policies.GPPSID
+	}
+
+	if request.IsBidRequest() && request.bidRequest.Regs != nil {
+		return request.bidRequest.Regs.GPPSID
+	}
+
+	return nil
+}
```
[the remark this answers is not in the document]
Comment (VeronikaSolovei9) [reply privacy/rules.go:90]: Yes! Looks good and simple to understand!

File: privacy/scrubber.go (hunk 2 above)
Comment (SyntaxNode) [reply privacy/scrubber.go:97]: > Maybe we should put it back?

Auto-completion works fine for me on VS Code with Go 1.20.5. Is there an update available for IntelliJ?

File: privacy/activity.go (hunk 3 above)
Comment (SyntaxNode) [reply privacy/activity.go:15]: Not a problem. Just being picky about honoring Go acronym conventions (my IDE "yelled" at me here too).

File: privacy/rules.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (SyntaxNode) [reply privacy/rules.go:77]: Sure. I think that would be good to add with Fetchers 2.0 so the map can be cached.

File: privacy/rules.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (SyntaxNode) [reply privacy/rules.go:90]: Yay!

File: privacy/scrubber.go (hunk 2 above)
Comment (VeronikaSolovei9) [reply privacy/scrubber.go:97]: Just updated. It didn't help. 
This is just one line of code that doesn't take memory and CPU, but makes IDEs happy. 
> Maybe we should put it back? 

Screenshot 2023-09-06 at 11 53 51 AM

Comment (SyntaxNode) [reply privacy/scrubber.go:97]: Let's make a deal. I'll revert and could you please open a bug report for IntelliJ?

**Review by VeronikaSolovei9 — DISMISSED** [review]

File: privacy/rules.go [hunk 6]
```diff
@@ -23,16 +27,18 @@ func (r ComponentEnforcementRule) Evaluate(target Component) ActivityResult {
 		return ActivityAbstain
 	}
 
+	if matched := evaluateGPPSID(r.gppSID, request); !matched {
+		return ActivityAbstain
```
Comment (AlexBVolcy) [inline privacy/rules.go:31]: I noticed this return value isn't covered by tests. Just want to point it out.

Comment (SyntaxNode) [reply privacy/rules.go]: Test added.

**Review by hhhjort — APPROVED** [review]
LGTM

**Review by VeronikaSolovei9 — APPROVED** [review]

**Pull request merged**